app/services/force/listing_service.rb

```diff
@@ -32,7 +32,8 @@ class ListingService
     # `ids` is a comma-separated list of ids
     # returns cached and cleaned listings
     def self.listings(attrs = {})
-      params = attrs[:ids].present? ? { ids: attrs[:ids] } : nil
+      params = { Tenure: attrs[:Tenure] }
+      params[:ids] = { ids: attrs[:ids] } if attrs[:ids].present?
       results = get_listings(params)
```

Comment:
`get_listings` calls `cached_get` which is contains this code:

```
      @cache.fetch(key, force: force_refresh, expires_in: expires_in) do
        get(endpoint, params)
      end
```

It asks Rails cache for the corresponding key, which is defined as:

```
      key = "#{endpoint}#{params ? '?' + params.to_query : ''}"
```

The code change above allows the `Tenure` attrs to be part of the key params so that the cache can return data saved to the key `/ListingDetails?Tenure=sale` and `/ListingDetails?Tenure=rental`.

In line 45, `ListingService#raw_listings`, is the method that's called in the caching rake task, so I updated that to cache for rental listings and sale listings _separately_.

If you access the rails console for this PR review app and run `Rails.cache.fetch('/ListingDetails?Tenure=sale')` and `Rails.cache.fetch('/ListingDetails?Tenure=rental')`, you'll see the cached data saved to each key respectively :)

To access the rails console for this review app, you can locally run `heroku run rails c --app dahlia-full-pr-1211`.